### DISCUSSION TITLE
Error in TypeSpec Code: duplicate-property: Model already has a property named

### DISCUSSION THREAD

**Discussion by yasu7ri** [microsoft/typespec#3531] [Q&A] [answered]
When I write the code as below, I get an error `duplicate-property: Model already has a property named countryCode`. 
Why is this happening?

```typescript
import "@typespec/http";

using TypeSpec.Http;

@tag("Country")
@route("/countries")
interface Countries {
    @route("{countryCode}/update") @post update(@path countryCode: string, ...Country): void | Error;
}

@doc("Country")
model Country {
    @doc("countryCode")
    countryCode: string;
    @doc("countryName")
    countryName: string;
}

```

```
countries.tsp:8:76 - error duplicate-property: Model already has a property named countryCode
> 8 |     @route("{countryCode}/update") @post update(@path countryCode: string, ...Country): void | Error;
    |                                                                            ^^^^^^^^^^

Found 1 error.
```

Is it not possible to use a property name from the model as a path parameter?

**Comment by timotheeguerin** [comment, marked as answer]
Depending on what you exactly want to do here, there is 2 options
```ts
import "@typespec/http";

using TypeSpec.Http;

// Option 1 - here you reuse the same property
@route("1/{countryCode}/update") op read1(...Country1): void;

model Country1 {
  @path countryCode: string;
  countryName: string;
}

// Option 2 - here the country is just the body and you still have an explicit path param 
@route("2/{countryCode}/update") op read2(
  @path countryCode: string,
  @body country: Country2,
): void;

model Country2 {
  countryCode: string;
  countryName: string;
}
```
[Playground example](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwOwoKLy8gT3B0aW9uIDEKQHJvdXRlKCIxL3tjb3VudHJ5Q29kZX0vdXBkYXRlIikgb3AgcmVhZDEoLi4uQ8YiMSk6IHZvaWQ7Cgptb2RlbCDIGCB7CiAgQHBhdGggy006IHN0cmluZzsKIMgXTmFtyxd97ACTMukAkzL%2FAJMyKNxuLMQdYm9kecgdOugApzIsCvcAwTLlAMH%2FALvvALs%3D&e=%40typespec%2Fopenapi3&options=%7B%7D)

**Reply by yasu7ri** [reply]
Thank you for your response.

I wanted to use the country model in the body, so I will use Option 2, which specifies @body as a parameter.
